community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogNeoCommandReaderV2_1.java

```diff
@@ -309,7 +310,7 @@ public boolean visitRelationshipGroupCommand( Command.RelationshipGroupCommand c
             {
                 throw new IOException( "Illegal in use flag: " + inUseByte );
             }
-            int type = channel.getShort();
+            int type = shortToUnsignedInt( channel.getShort() );
```

Comment:
But since we do not support upgrades for stores that was shutdown in a non clean way do we need to touch those readers?